tests/test_decompositions.py

```diff
@@ -145,6 +145,21 @@ def test_clements_phase_end_random_unitary(self):
             U_rec = np.diag(new_diags) @ U_rec
             self.assertAlmostEqual(np.linalg.norm(U_rec-U), 0, delta=self.tol)
 
+    def test_triangular_decomposition_random_unitary(self):
```

Comment:
@ishdhand, could you add a (one line) docstring here briefly explaining what this test is checking, and the expected result? This will show up during failures in our test runner, so will be useful for future debugging.

Reply:
Thanks @josh146 for going through the code! I have added the docstrings in ede5be37448872eec4efeb860e4ebc46fb95bc26

Reply:
Perfect, looks great @ishdhand